lib/shared/addon/components/cluster-driver/driver-linodelke/component.js

```diff
@@ -0,0 +1,360 @@
+            k8sVersions: responses.k8sVersions.data,
+          });
+          cb(true);
+        }).catch((err) => {
+          if (err && err.body && err.body.errors && err.body.errors[0]) {
+            errors.push(`Error received from Linode: ${ err.body.errors[0].reason }`);
+          } else {
+            errors.push(`Error received from Linode`);
+          }
+
+          this.setProperties({ errors, });
+          cb(false);
+        });
+      }
+    },
+    verifyClusterConfig(cb) {
+      // verify if tags are not null
+      // if null replace with empty array
+      const tags = get(this, 'cluster.lkeEngineConfig.tags');
+
+      if (!tags) {
+        set(this, 'cluster.lkeEngineConfig.tags', []);
+      }
+      set(this, 'step', 3);
+      cb(true);
+    },
+    createCluster(cb) {
+      if (this.verifyNodePoolConfig()) {
+        this.send('driverSave', cb);
+      } else {
+        cb(false);
+      }
+    },
+
+    updateCluster(cb) {
+      if (this.verifyNodePoolConfig()) {
+        this.send('driverSave', cb);
+      } else {
+        cb(false);
+      }
+    },
+
+    cancelFunc(cb){
+      // probably should not remove this as its what every other driver uses to get back
+      get(this, 'router').transitionTo('global-admin.clusters.index');
+      cb(true);
+    },
+
+    // for tags
+    addNewTag() {
+      const tags = get(this, 'cluster.lkeEngineConfig.tags') || [];
+      const newTag = get(this, 'newTag');
+
+      if (newTag) {
+        tags.pushObject(newTag);
+        set(this, 'cluster.lkeEngineConfig.tags', tags);
+        set(this, 'newTag', '');
+      }
+    },
+    deleteTag(idx) {
+      const tags = get(this, 'cluster.lkeEngineConfig.tags') || [];
+
+      set(this, 'cluster.lkeEngineConfig.tags', tags.filter((tag, index) => index !== idx));
+    },
+
+    // for node pools
+    addSelectedNodePool() {
+      const selectedNodePoolObj = get(this, 'selectedNodePoolObj');
+      const selectedNodePoolList = get(this, 'selectedNodePoolList');
+
+      if (selectedNodePoolObj.id) {
+        // add to list
+        selectedNodePoolList.pushObject(selectedNodePoolObj);
+
+        // clear selected
+        set(this, 'selectedNodePoolType', '');
+        set(this, 'selectedNodePoolObj', {});
+      }
+    },
+    deleteNodePool(id) {
+      const selectedNodePoolList = get(this, 'selectedNodePoolList');
+
+      set(this, 'selectedNodePoolList', selectedNodePoolList.filter((n) => n.id !== id))
+    }
+  },
+
+
+  // For languages
+  languageChanged: observer('intl.locale', function() {
+    const lang = get(this, 'intl.locale');
+
+    if (lang) {
+      this.loadLanguage(lang[0]);
+    }
+  }),
+  clusterNameChanged: observer('cluster.name', function() {
+    const clusterName = get(this, 'cluster.name');
+
+    set(this, 'cluster.lkeEngineConfig.name', clusterName);
+    set(this, 'cluster.lkeEngineConfig.label', clusterName);
```

Comment:
`setProperties` is especially useful when working with observers as setProperties will only notify watchers after all properties have been set vs each set as implemented.